pyccel/stdlib/ndarrays/ndarrays.c

```diff
@@ -373,3 +429,65 @@ int64_t     *numpy_to_ndarray_shape(int64_t *np_shape, int nd)
     return nd_shape;
 
 }
+
+bool    is_same_shape(t_ndarray dest, t_ndarray src)
+{
+    int i = 0;
+
+    if (dest.nd != src.nd)
+        return false;
+    while (i < dest.nd)
+    {
+        if (dest.shape[i] != src.shape[i])
+            return false;
+        ++i;
+    }
+    return true;
+}
+
+int get_shape_product(int64_t *shape, int nd, int max_nd)
+{
+    int product = 1;
+
+    for (int i = nd; i < max_nd; ++i)
+        product *= shape[i];
+    return (product);
+}
+
+int index_position(t_ndarray arr, int index, int nd)

```

Comment:
Where is this function used? Isn't `index` already the output of a function? Would it not make more sense to simply correct that function?

Reply:
They have a different purpose, one uses `va_args` to translate from `(array[0][0])` to its one dimensional value `(0)`,
example `array([[1, 2, 3], [4, 5, 6]]` the `get_index(array, 0, 1)` would return `3` for `order_c` since `order_c`'s memory layout would be `[1][2][3][4][5][6]` and `1` for `order_f`
since `order_f`'s memory layout would be `[1][4][2][5][3][6]`.
`index_position` returns the index of the 'Nth' element where`N` >= 0:
for `array([[1, 2, 3], [4, 5, 6]]` `order_f`, to get the third element, `index_position(arr, 3, 2)` would return '1'.
for `array([[1, 2, 3], [4, 5, 6]]` `order_c`, to get the third element, `index_position(arr, 3, 2)` would return '3'.
for `order_c` it always returns `index`.
this works for both slicing and different order copying, this is actually just a refactoring of @soukainahikma's function.
i can see that the naming is not good, this might also be removed after we discuss how ordering should be handled.